### PR TITLE
Add QfDialog.

### PR DESCRIPTION
**Pull request by mohsenD98**
Cleaner dialog codes + we can have our custom dialog later. remove lots of duplication.

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 2fd6103b635249f4783b365110b580b5f7e9ac09: [**arm64-android**](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5733-c79c5d48c4f7be48b3cf891734c4ac68fe143b73-android-arm64.apk)
<details>
<summary>Other architectures</summary>

- [arm-neon-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5733-c79c5d48c4f7be48b3cf891734c4ac68fe143b73-android-arm-neon.apk)
- [x64-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5733-c79c5d48c4f7be48b3cf891734c4ac68fe143b73-android-x64.apk)
- [x86-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5733-c79c5d48c4f7be48b3cf891734c4ac68fe143b73-android-x86.apk)
- [**all access arm64-android**](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5733-c79c5d48c4f7be48b3cf891734c4ac68fe143b73-android-arm64.apk)
- [all access arm-neon-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5733-c79c5d48c4f7be48b3cf891734c4ac68fe143b73-android-arm-neon.apk)
- [all access x64-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5733-c79c5d48c4f7be48b3cf891734c4ac68fe143b73-android-x64.apk)
- [all access x86-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5733-c79c5d48c4f7be48b3cf891734c4ac68fe143b73-android-x86.apk)
</details>

File: src/qml/FeatureListForm.qml [hunk 1]
```diff
@@ -911,28 +894,15 @@ Rectangle {
...
-
     title: qsTr("Delete feature(s)")
-    Label {
-      width: parent.width
-      wrapMode: Text.WordWrap
-      text: qsTr("Should the %n feature(s) selected really be deleted?", "0", deleteDialog.selectedCount)
-    }
-
-    standardButtons: Dialog.Ok | Dialog.Cancel
+    text: qsTr("Should the %n feature(s) selected really be deleted?", "0", deleteDialog.selectedCount)
```
Comment (nirvn) [inline src/qml/FeatureListForm.qml:905]: Why did you get rid of Label {} here?

File: src/qml/PluginManagerSettings.qml [hunk 2]
```diff
@@ -301,17 +297,16 @@ Popup {
...
     }
   }
 
-  Dialog {
+  QfDialog {
     id: uninstallConfirmation
     title: "Uninstall Plugin"
     parent: mainWindow.contentItem
-    x: (mainWindow.width - width) / 2
+    modal: false
```
Comment (nirvn) [inline src/qml/PluginManagerSettings.qml:309]: I don't think we need modal: false here; the lack of model: true was most likely me being forgetful.

Comment (mohsenD98) [reply src/qml/PluginManagerSettings.qml:309]: Agreed.

**Review by nirvn — APPROVED** [review]

**Pull request merged**